src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -274,13 +273,11 @@ private static void EmitFindFirstChar(IndentedTextWriter writer, RegexMethod rm,
             bool hasTextInfo = false;
 
             // In some cases, we need to emit declarations at the beginning of the method, but we only discover we need them later.
-            // To handle that, we emit a placeholder value that's not valid C#, and then at the end of the code generation we either
-            // delete it if no additional declarations are required, or we replace it with the list of additional declarations
-            // built up while generating code.
+            // To handle that, we build up a collection of all the declarations to include, track where they should be inserted,
+            // and then insert them at that position once everything else has been output.
             var additionalDeclarations = new HashSet<string>();
 
             // Emit locals initialization
-            writer.WriteLine("global::System.ReadOnlySpan<char> inputSpan = base.runtext;");
```

Comment:
So this is only for `FindNextStartingPositionMode.NoSearch` right? Just for my understanding, can you provide a pattern and input text that would trigger such a case as I'm not sure what that means?

Reply:
oh, I guess an empty pattern would do this since there is nothing really to compare 😄. Disregard my question.